onmt/train_single.py

```diff
@@ -67,6 +77,7 @@ def _init_train(opt):
             if len(old_transf) != 0:
                 _msg += f" -{old_transf}."
             logger.warning(_msg)
+            vocabs, transforms_cls = prepare_transforms_vocabs(opt)
```

Comment:
Not sure why this was not done before. Also, since it's a duplicate of line 83, and 86, it can probably be put only once after the conditions.

Reply:
TBH even with this, the logic is not 100% bullet proof. But it helps. I will re-order things.

Reply:
Well there is a case:
if you want to do a plain train_from and just take the vocab options from the checkpoint, since the model_opt is loaded after, you need to keep the vocab from the checkpoint and not go through the prepare_transforms_vocabs(opt).
This is why I am saying the logic is not 100% bullet proof. I added the line in the case of transforms changes but technically it also requires the opt to define the vocab.